Source/JavaScriptCore/runtime/ProgramExecutable.cpp

```diff
@@ -94,8 +94,7 @@ JSObject* ProgramExecutable::initializeGlobalProperties(VM& vm, JSGlobalObject*
     UnlinkedProgramCodeBlock* unlinkedCodeBlock = vm.codeCache()->getUnlinkedProgramCodeBlock(
         vm, this, source(), codeGenerationMode, error);
 
-    if (globalObject->hasDebugger())
-        globalObject->debugger()->sourceParsed(globalObject, source().provider(), error.line(), error.message());
+    globalObject->sourceParsed(source().provider(), error.line(), error.message());
```

Comment:
Always asking to the hashtable is too costly. You need to come up with an idea not doing that when debugger is disabled.